uxarray/grid/bounds.py

```diff
@@ -0,0 +1,559 @@
+    """
+    grid.normalize_cartesian_coordinates()
+
+    bounds_array = _construct_face_bounds(
+        grid.face_node_connectivity.values,
+        grid.n_nodes_per_face.values,
+        grid.node_x.values,
+        grid.node_y.values,
+        grid.node_z.values,
+        grid.node_lon.values,
+        grid.node_lat.values,
+        is_latlonface,
+        is_face_GCA_list,
+    )
+
+    bounds_da = xr.DataArray(
+        bounds_array,
+        dims=["n_face", "lon_lat", "min_max"],
+        attrs={
+            "cf_role": "face_latlon_bounds",
+            "_FillValue": INT_FILL_VALUE,
+            "long_name": "Latitude and longitude bounds for each face in radians.",
+        },
+    )
+
+    if process_intervals:
+        intervals_tuple_list = []
+        intervals_name_list = []
+        for face_idx in range(grid.n_face):
+            assert bounds_array[face_idx][0][0] != bounds_array[face_idx][0][1]
+            assert bounds_array[face_idx][1][0] != bounds_array[face_idx][1][1]
+            lat_array = bounds_array[face_idx][0]
+            intervals_tuple_list.append((lat_array[0], lat_array[1]))
+            intervals_name_list.append(face_idx)
+
+        intervalsIndex = pd.IntervalIndex.from_tuples(
+            intervals_tuple_list, closed="both"
+        )
+        df_intervals_map = pd.DataFrame(
+            index=intervalsIndex, data=intervals_name_list, columns=["face_id"]
+        )
+        bounds_da.assign_attrs(
+            {
+                "latitude_intervalsIndex": intervalsIndex,
+                "latitude_intervals_name_map": df_intervals_map,
+            }
+        )
+
+    if return_array:
+        return bounds_da
+    else:
+        grid._ds["bounds"] = bounds_da
+
+
+@njit(cache=True, parallel=True)
+def _construct_face_bounds(
+    face_node_connectivity,
+    n_nodes_per_face,
+    node_x,
+    node_y,
+    node_z,
+    node_lon,
+    node_lat,
+    is_latlonface: bool = False,
+    is_face_GCA_list=None,
+):
+    """Computes an array of face bounds."""
+    n_face = face_node_connectivity.shape[0]
+    bounds_array = np.empty((n_face, 2, 2), dtype=np.float64)
+
+    node_lon = np.deg2rad(node_lon)
+    node_lat = np.deg2rad(node_lat)
+
+    # Iterate over each face in parallel
+    for face_idx in prange(n_face):
+        # 1) Create the Cartesian Face Edge Nodes for the current face
+        cur_face_edge_nodes_cartesian = _edge_nodes_cartesian_kernel(
+            face_idx, face_node_connectivity, n_nodes_per_face, node_x, node_y, node_z
+        )
+        # 2) Create the Spherical Face Edge Nodes for the current face
+        cur_face_edge_nodes_spherical = _edge_nodes_spherical_kernel(
+            face_idx, face_node_connectivity, n_nodes_per_face, node_lon, node_lat
+        )
+
+        # TODO: This isn't currently used for the user API, consider updating in the future
```

Comment:
Our documentation stated that we support either great circle arcs or constant latitude. (This is also what YAC/CDO supports) And we should probably consider officially support that in our APIs in the future

Reply:
Agreed.